backend/src/main/java/ca/bc/gov/backendstartapi/exception/ActivityNotFoundException.java

```diff
@@ -0,0 +1,9 @@
+package ca.bc.gov.backendstartapi.exception;
+
+/** This class represents a not found activity from ActivityEnum. */
+public class ActivityNotFoundException extends RuntimeException {
```

Comment:
One suggestion is to always make your exceptions extend from `ResponseStatusException` and add the `@ResponseStatus` annotation to the class to make spring identify and throw the correct error message without depending too much on specific error handler methods.

Example:
```java
@ResponseStatus(value = HttpStatus.NOT_FOUND)
public class ActivityNotFoundException extends ResponseStatusException {
  public ActivityNotFoundException() {
    super(HttpStatus.NOT_FOUND, "Activity don't exist!");
  }
}
```

Reply:
Hmm. That is an interesting idea. Maybe we could catch the annotated value in the `RestControllerAdvice` and return it. What do you think?

Reply:
Right. I did better. Sending changes soon!